src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,567 @@
+                                        summary='Error while doing helm get values azure-arc')
+                raise CLIInternalError(str.format(consts.Update_Agent_Failure, error_helm_get_values.decode("ascii")))
+
+        # Converting output obtained in json format and fetching the clusterconnect-agent feature
+        helm_values_json = json.loads(output_helm_values_get)
+        cluster_connect_feature = helm_values_json["systemDefaultValues"]["clusterconnect-agent"]["enabled"]
+
+        # To retrieve all of the arc agent pods that are presesnt in the Cluster
+        arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+        # Traversing thorugh all agents and checking if the Kube aad proxy pod is present or not
+        for each_agent_pod in arc_agents_pod_list.items:
+
+            if(each_agent_pod.metadata.name.startswith("kube-aad-proxy")):
+                kap_pod_present = True
+                break
+
+        # Checking if any pod security policy is set
+        if(cluster_connect_feature is True and kap_pod_present is False):
+            print("Error: Unable to create Kube-aad-proxy proxy deployment there is some network policy present in the cluster.\n Refer this link to for more information 'security_policy_link'.\n")
+            return False
+        return True
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to check for the presence of cluster security policy in the cluster. Exception: {}".format(str(e)) + "\n")
+
+
+def check_kap_cert(corev1_api_instance):
+
+    try:
+        # Initialize the kap_cert_present as False
+        kap_cert_present = False
+        kap_pod_status = ""
+
+        # To retrieve all of the arc agent pods that are presesnt in the Cluster
+        arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+        # Traversing thorugh all agents and checking if the Kube aad proxy pod is in containercreating state
+        for each_agent_pod in arc_agents_pod_list.items:
+
+            if each_agent_pod.metadata.name.startswith("kube-aad-proxy") and each_agent_pod.status.phase == "ContainerCreating":
+                kap_pod_status = "ContainerCreating"
+                break
+
+        # Going thorugh all the secrets in azure-arc
+        all_secrets_azurearc = corev1_api_instance.list_namespaced_secret(namespace="azure-arc")
+
+        for secrets in all_secrets_azurearc.items:
+
+            # If name of secret is kube-aad-proxy-certificate then we stop there
+            if(secrets.metadata.name == "kube-aad-proxy-certificate"):
+                kap_cert_present = True
+
+        if not kap_cert_present and kap_pod_status == "ContainerCreating":
+            print("Error: Unable to pull Kube aad proxy certificate.\n Refer this link to for more information 'kap_cert_link'. \n")
```

Comment:
Instead of refer to this link, we can ask the customer to retry onboarding - Adding @krdhruva to take a look at the error message

Reply:
Also, let's ensure we're capturing this CLI output also to the local machine's diagnostic_logs folder

Reply:
Added the code to capture the CLI output.